### PR TITLE
(Wallet) Implement onboarding initial screen and TOC screen

### PR DESCRIPTION
**Pull request by simoarpe**
Screenshot 2024-02-05 at 5 48 22 PM

Screenshot 2024-02-05 at 5 48 41 PM

## Resolves https://github.com/brave/brave-browser/issues/35713

Implements new initial onboarding screen and new terms of use screen.

### Light Mode

image

image

### Dark Mode

image

image

Figma design: https://www.figma.com/file/oXIiv45pIhF2RGlpEjOfUg/Wallet-onboarding?node-id=6223%3A70942&mode=dev

Main umbrella issue: https://github.com/brave/brave-browser/issues/35294

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [X] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

* Erase any wallet data, if any.
* Navigate to Wallet section
* Observe new onboarding creation screen.
* Tap on "Need a new Wallet".
* Observe new terms of use screen.
* Verify "continue" button is enabled only when both checkboxes are ticked
* Verify "Terms of use" clickable span points to https://brave.com/terms-of-use/

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`strings`++

File: android/java/res/layout/fragment_setup_wallet.xml [hunk 1]
```diff
@@ -15,80 +15,178 @@
...
+            <RelativeLayout
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:layout_margin="32dp">
+                <ImageView
+                    android:id="@+id/restore_wallet_icon"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:contentDescription="@null"
+                    android:src="@drawable/restore_wallet"
```
Comment (SergeyZhukovsky) [inline android/java/res/layout/fragment_setup_wallet.xml:118]: I remember we had problems with `src` vector images on Android below 7 and we used `srcCompat`. From what I'm reading right now it also supports scaling. What do you think is it beneficial to use `srcCompat`. I would probably use it everywhere.

File: android/java/res/layout/fragment_setup_wallet.xml [hunk 2]
```diff
@@ -15,80 +15,178 @@
...
+                    android:layout_toEndOf="@+id/restore_wallet_icon"
+                    android:layout_below="@+id/restore_text_view"
+                    />
+
+                <ImageView
+                    android:id="@+id/brave_icon"
+                    android:layout_width="24dp"
+                    android:layout_height="24dp"
+                    android:contentDescription="@null"
+                    android:src="@drawable/ic_brave"
```
Comment (SergeyZhukovsky) [inline android/java/res/layout/fragment_setup_wallet.xml:154]: I would check all `src` if we want to go with `srcCompat`

File: android/java/res/layout/fragment_setup_wallet.xml (hunk 1 above)
Comment (simoarpe) [reply android/java/res/layout/fragment_setup_wallet.xml:118]: `support-vector-drawable` library was fully integrated in Android 7.0 (API level 24): https://developer.android.com/develop/ui/views/graphics/vector-drawable-resources#vector-drawables-backward-solution.
The code used for rendering vector drawables from `src` and `srcCompat` should be unified now. 
Our minimum SDK is now Android Oreo (API 26), so we should be super safe here.
I also didn't notice any distortion, here is a screenshot:

Screenshot 2024-02-05 at 5 48 22 PM

File: android/java/res/layout/fragment_setup_wallet.xml (hunk 2 above)
Comment (simoarpe) [reply android/java/res/layout/fragment_setup_wallet.xml:154]: If you have a preference over `srcCompat` I'm happy to change all the occurrences.

Comment (SergeyZhukovsky) [reply android/java/res/layout/fragment_setup_wallet.xml:154]: no, I'm good, just wanted to be extra careful that we can use it now. I'm totally fine to have `src`.

**Pull request merged**